src/Sarif.Viewer.VisualStudio/CodeFlowToTreeConverter.cs

```diff
@@ -27,14 +29,24 @@ private static List<CallTreeNode> GetChildren(CodeFlow codeFlow, ref int current
                 switch (codeFlow.Locations[currentCodeFlowIndex].Kind)
                 {
                     case AnnotatedCodeLocationKind.Call:
+                        unReturnedCalls.Push(codeFlow.Locations[currentCodeFlowIndex]);
                         children.Add(new CallTreeNode
                         {
                             Location = codeFlow.Locations[currentCodeFlowIndex],
-                            Children = GetChildren(codeFlow, ref currentCodeFlowIndex)
+                            Children = GetChildren(codeFlow, ref currentCodeFlowIndex, ref unReturnedCalls)
                         });
                         break;
 
                     case AnnotatedCodeLocationKind.CallReturn:
+                        if (unReturnedCalls.Count == 0)
+                        {
+                            throw new System.ArgumentException("At least one AnnotatedCodeLocation Call in this CodeFlow is not paired with a CallReturn, causing an imbalanced tree.");
```

Comment:
InvalidDataException might be better. ... although as Larry stated, the SARIF is valid.